linuxPackages.librem-ec-acpi: init

sjamaan opened the pull request:
This packages the [librem-ec-acpi-dkms](https://source.puri.sm/nicole.faerber/librem-ec-acpi-dkms) kernel module.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

sjamaan commented:
NOTE: This is basically an updated version of #134856, bumped upstream to use the v0.9.2 branch and renamed the package to `librem-ec-acpi` (dropped the `-dkms` suffix as suggested by @K900)

sjamaan commented on pkgs/os-specific/linux/purism/librem-ec-acpi/default.nix:
Done!

FliegendeWurst commented:
@ofborg build linuxKernel.packages.linux_6_6.librem-ec-acpi

FliegendeWurst commented:
LGTM. Would be good to squash the commits.

sjamaan commented:
> LGTM. Would be good to squash the commits.

Done

me-and commented on pkgs/os-specific/linux/purism/librem-ec-acpi/default.nix:
```diff
@@ -0,0 +1,41 @@
+  meta = with lib; {
+    homepage = "https://source.puri.sm/nicole.faerber/librem-ec-acpi-dkms";
+    maintainers = [ maintainers.sjamaan ];
+    license = lib.licenses.gpl2;
```
`lib.licenses.gpl2` is deprecated; this should be `lib.licenses.gpl2Only`.

me-and commented on pkgs/os-specific/linux/purism/librem-ec-acpi/default.nix:
```diff
@@ -0,0 +1,41 @@
+  installFlags = [ "DEPMOD=true" ];
+  enableParallelBuilding = true;
+
+  patches = [ ./Makefile.patch ];
```
This should either have a comment describing either the upstream URL, or a reason why the patch wasn't upstreamed.

me-and commented on pkgs/os-specific/linux/purism/librem-ec-acpi/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  buildFlags = [ "all" ];
```
Nit: this seems unnecessary, as `all` is the default build target already.

me-and commented on pkgs/os-specific/linux/purism/librem-ec-acpi/default.nix:
```diff
@@ -0,0 +1,41 @@
+    owner = "nicole.faerber";
+    repo = "librem-ec-acpi-dkms";
+    rev = "cec8c0e8bf1532c9c605f60bb02a2ef0f98e5d77";
+    sha256 = "gPb1/4UPbwjJiqtP27KKjiIKxIYEl2nlQACUAlxVEFU=";
```
Nit: per https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-fetchers-fetchurl-inputs it's generally preferred to provide this as a `hash` attribute.

me-and commented on pkgs/os-specific/linux/purism/librem-ec-acpi/default.nix:
```diff
@@ -0,0 +1,41 @@
+  kernel,
+}:
+
+stdenv.mkDerivation rec {
```
Nit: I'd drop the `rec` here, as I don't think there's currently anything recursive about your definition.

me-and commented on pkgs/os-specific/linux/purism/librem-ec-acpi/default.nix:
```diff
@@ -0,0 +1,41 @@
+    domain = "source.puri.sm";
+    owner = "nicole.faerber";
+    repo = "librem-ec-acpi-dkms";
+    rev = "cec8c0e8bf1532c9c605f60bb02a2ef0f98e5d77";
```
Nit: this is the commit hash for the tag, but it'd be clearer to just reference the tag name in my opinion. That'd probably mean you'd want something like

```nix
let
  version = "0.9.2";
in
stdenv.mkDerivation {
  version = "${version}-${kernel.version}";
  src = fetchFromGitLab {
    rev = "v${version}";
    ...
  };
  ...
}
```

me-and reviewed (changes requested):
Picked a random PR to review. Overall looks good, although I think there's a couple of easy-to-fix changes needed to get this over the line.

### New maintainer

- [x] No GPG key, or GPG key is used to sign the commit(s)
- [x] GitHub account name and ID number match

### New package

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux *(kernel module so nothing to test)*
- [x] `meta.description` is set and fits guidelines
- [ ] `meta.license` fits upstream license *(see review comment)*
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] `meta.mainProgram` is set, if applicable.
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [ ] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed *(see review comment)*
- [x] patches that are remotely available are fetched rather than vendored

me-and commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352070`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.librem-ec-acpi</li>
    <li>linuxPackages_5_4_hardened.librem-ec-acpi</li>
  </ul>
</details>
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_6_11.librem-ec-acpi</li>
    <li>linuxPackages_6_11_hardened.librem-ec-acpi (linuxKernel.packages.linux_6_11_hardened.librem-ec-acpi)</li>
    <li>linuxPackages_latest-libre.librem-ec-acpi (linuxKernel.packages.linux_latest_libre.librem-ec-acpi)</li>
    <li>linuxPackages_lqx.librem-ec-acpi (linuxKernel.packages.linux_lqx.librem-ec-acpi)</li>
    <li>linuxPackages_xanmod_latest.librem-ec-acpi (linuxKernel.packages.linux_xanmod_latest.librem-ec-acpi ,linuxPackages_xanmod_stable.librem-ec-acpi)</li>
    <li>linuxPackages_zen.librem-ec-acpi (linuxKernel.packages.linux_zen.librem-ec-acpi)</li>
    <li>linuxPackages_latest.librem-ec-acpi</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.librem-ec-acpi</li>
    <li>linuxPackages_5_10_hardened.librem-ec-acpi (linuxKernel.packages.linux_5_10_hardened.librem-ec-acpi)</li>
    <li>linuxKernel.packages.linux_5_15.librem-ec-acpi</li>
    <li>linuxPackages_5_15_hardened.librem-ec-acpi (linuxKernel.packages.linux_5_15_hardened.librem-ec-acpi)</li>
    <li>linuxKernel.packages.linux_5_4.librem-ec-acpi</li>
    <li>linuxKernel.packages.linux_6_1.librem-ec-acpi</li>
    <li>linuxPackages_6_1_hardened.librem-ec-acpi (linuxKernel.packages.linux_6_1_hardened.librem-ec-acpi)</li>
    <li>linuxPackages.librem-ec-acpi (linuxKernel.packages.linux_6_6.librem-ec-acpi)</li>
    <li>linuxPackages_hardened.librem-ec-acpi (linuxPackages_6_6_hardened.librem-ec-acpi)</li>
    <li>linuxPackages-libre.librem-ec-acpi (linuxKernel.packages.linux_libre.librem-ec-acpi)</li>
    <li>linuxPackages_xanmod.librem-ec-acpi (linuxKernel.packages.linux_xanmod.librem-ec-acpi)</li>
  </ul>
</details>

sjamaan commented:
Thanks for the review! I've made some changes to address your feedback. Please have another look.

FliegendeWurst commented on maintainers/maintainer-list.nix:
```diff
@@ -21234,6 +21234,11 @@
     githubId = 80012;
     name = "Simon Jagoe";
   };
+  sjamaan = {
```
This needs to be a separate commit.

sjamaan replied:
Done!

Unfortunately, splitting the commit caused the author info of the original commit to be lost.

sjamaan committed:
Add sjamaan to maintainers list

sjamaan committed:
librem-ec-acpi: init at 0.9.2

Users of NixOS on the Librem 14 laptop from Purism will want to include
this out-of-tree kernel module in their system configuration

  boot.extraModulePackages = with config.boot.kernelPackages;
    [ librem-ec-acpi ];

me-and commented:
Changes look good, although I've not yet tried rebuilding since the failed builds last time around...

me-and commented:
...huh, `nixpkgs-review pr 352070` is now reporting "Nothing to be built." I think that must be a lie, given, for example, `linuxPackages_latest.librem-ec-acpi` exists with this code but not without it.

Nonetheless, I am still getting build failures when I try to build `linuxPackages_latest.librem-ec-acpi`.

sjamaan commented:
> ...huh, `nixpkgs-review pr 352070` is now reporting "Nothing to be built." I think that must be a lie, given, for example, `linuxPackages_latest.librem-ec-acpi` exists with this code but not without it.
> 
> Nonetheless, I am still getting build failures when I try to build `linuxPackages_latest.librem-ec-acpi`.

I noticed there's [an upstream MR](https://source.puri.sm/nicole.faerber/librem-ec-acpi-dkms/-/merge_requests/2) which addresses a build failure on Linux 6.10.3. Is this the error you see?

me-and commented:
Yes, that looks like the problem!

```
/build/source/librem_ec_acpi.c:782:10: error: 'struct acpi_driver' has no member named 'owner'
  782 |         .owner = THIS_MODULE,
      |          ^~~~~
```

FliegendeWurst commented:
In that case the linked patch should be included here.

FliegendeWurst commented on pkgs/os-specific/linux/purism/librem-ec-acpi/default.nix:
```diff
@@ -0,0 +1,49 @@
+    maintainers = [ maintainers.sjamaan ];
+    license = lib.licenses.gpl2Only;
+    description = "Kernel module for the Purism Librem EC ACPI DKMS";
+    platforms = platforms.linux;
```
```suggestion
    platforms = [ "x86_64-linux" ];
```
The hardware is only made for x86

FliegendeWurst commented:
Please also squash the last three commits so you just have:
- `maintainers: add sjamaan`
- `linuxPackages.librem-ec-acpi: init at <version>` (with correct version)

sjamaan commented:
Sorry, was sick for a while and didn't have the energy to pick this back up. Should be good now
